tls/s2n_client_cert_preferences.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "stuffer/s2n_stuffer.h"
+#include "error/s2n_errno.h"
+#include "tls/s2n_client_cert_preferences.h"
+#include "utils/s2n_safety.h"
+
+int s2n_recv_client_cert_preferences(struct s2n_stuffer *in, s2n_cert_type *chosen_cert_type_out)
+{
+    uint8_t cert_types_len;
+    GUARD(s2n_stuffer_read_uint8(in, &cert_types_len));
+    GUARD(s2n_choose_preferred_client_cert_type(in, cert_types_len, chosen_cert_type_out));
+
```

Comment:
Why read the length here if it is only used in the helper function? It seems like all of the functionality is in s2n_choose_preferred_client_cert_type and could be collapsed back into this function.